### PR TITLE
Refactor allExistingInstancesMatch

### PR DESCRIPTION
**Pull request by iaguis**
Addresing https://github.com/giantswarm/aws-operator/pull/19#discussion_r103201418

### CODE REVIEW

File: service/create/service_test.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+				},
+			},
+			state: EC2TerminatedState,
+			res:   false,
+		},
+	}
+
+	for i, tc := range tests {
+		res := allExistingInstancesMatch(tc.instances, tc.state)
+		if res != tc.res {
```
Comment (nhlfr) [inline service/create/service_test.go]: Could you use https://github.com/stretchr/testify? We already have it vendored and I used it for k8s client testing.

Comment (iaguis) [inline service/create/service_test.go]: sure

File: service/create/service_test.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+							},
+						},
+					},
+				},
+			},
+			state: EC2TerminatedState,
+			res:   true,
+		},
+		{
+			name: "Expect not terminated a terminated instance and a running instance in a single reservation",
```
Comment (xh3b4sd) [inline service/create/service_test.go]: `Expect not terminated a terminated` -> `Expect not terminated with a terminated`.

**Comment by iaguis** [comment]
Updated. Depends on #21

File: service/create/service_test.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+				},
+			},
+			state: EC2TerminatedState,
+			res:   false,
+		},
+	}
+
+	for _, tc := range tests {
+		res := allExistingInstancesMatch(tc.instances, tc.state)
+		if res != tc.res {
```
Comment (nhlfr) [inline service/create/service_test.go]: If you are using assert.Equal, you don't need this if statement

Comment (iaguis) [inline service/create/service_test.go]: right, silly me

**Pull request merged**